.werft/build.ts

```diff
@@ -585,7 +585,13 @@ async function registerK3sWsCluster(namespace: string, domain: string, pathToKub
     // When we deploy the ws proxy then certs are not generate. So the pods could go into a long waiting state without kubelet reattempting to mount the volume
     // So, we just restart the deployment for secrets to be mounted.
     shell.exec(`export KUBECONFIG=${pathToKubeConfigK3s} && kubectl rollout restart deploy ws-proxy -n ${namespace}`)
-    shell.exec(`export KUBECONFIG=${pathToKubeConfigK3s} && kubectl rollout status deploy ws-proxy -n ${namespace}`)
+    waitForDeploymentToSucceed(pathToKubeConfigK3s, "ws-proxy", namespace, "deploy")
+
+    // Following deployments must have succeede for us to register the cluster
+    waitForDeploymentToSucceed("", "ws-manager-bridge", namespace, "deploy")
+    waitForDeploymentToSucceed("", "ws-manager", namespace, "deploy")
+    waitForDeploymentToSucceed("", "server", namespace, "deploy")
+
     // explicitly wait for 15s assuming the k3s service would have picked the new ws-proxy pod
     await sleep(15000)
```

Comment:
is this still needed when we wait for all workloads explicitly?

Reply:
We wait for workloads but not for the service frontending the ws-proxy pod. Service takes few seconds before it reads the pods matching its selectors.